superset/connectors/base/models.py

```diff
@@ -313,7 +313,9 @@ def data_for_slices(  # pylint: disable=too-many-locals
                     metric_names.add(utils.get_metric_name(metric))
                     if utils.is_adhoc_metric(metric):
                         column_names.add(
-                            (metric.get("column") or {}).get("column_name")
+                            (
+                                (hasattr(metric, "get") and metric.get("column")) or {}
+                            ).get("column_name")
```

Comment:
On line 314 we're already calling a type guard making sure that the type is in fact `AdhocMetric`. So it appears there may be something wrong either in the declaration of the `AdhocMetric` type or the type guard. Can you post the full stack trace of the error you're seeing and see if we need to update either one of those instead?

Reply:
@villebro sorry for the late reply. As you mentioned correctly, I missed the check for `AdhocMetric`. The issue was not with `metric` param, rather with `column`. Added a check around that.